ui/components/multichain-accounts/multichain-account-menu/multichain-account-menu.tsx

```diff
@@ -143,13 +151,20 @@ export const MultichainAccountMenu = ({
       </Box>
       <Popover
         className="multichain-account-cell-popover-menu"
-        isOpen={isPopoverOpen}
+        isOpen={isOpen}
         position={PopoverPosition.LeftStart}
         referenceElement={popoverRef.current}
         matchWidth={false}
         borderRadius={BorderRadius.LG}
+        isPortal
+        preventOverflow
+        flip
+        onClickOutside={onToggle}
+        onPressEscKey={onToggle}
       >
-        <MultichainAccountMenuItems menuConfig={menuConfig} />
+        <ModalFocus restoreFocus initialFocusRef={popoverRef}>
+          <MultichainAccountMenuItems menuConfig={menuConfig} />
+        </ModalFocus>
```

Comment:
ModalFocus is an accessibility consideration to lock the focus to the menu. However nothing in this menu meets accessibility standards so we will have to update to correct semantic html in another PR. This is what navigating around with the keyboard looks like